### DISCUSSION TITLE
Display PDF files using browser

### DISCUSSION THREAD

**Discussion by rustemos** [community/community#19347] [Other Feature Feedback, Questions, & Ideas]
Hello,

Would it be possible to add an option to display PDF files using browser instead of github window?

Thank you,
Rustem

**Comment by Huseyin-Cinar** [comment]
Ways to view files from Github repos in browser;

- Use Github Codespaces and right click on the file and select "open in browser". (It probably won't be, as it's currently a beta feature.)
- You can view the repository you want using hosting such as Netlify, Github.io and deploy it here on the browser.
- Download the file to your computer and use the "open with browser" option :)

Hope I was helpful, Good Coding :)

**Reply by rustemos** [reply]
Thank you for your reply and suggestions. It seems that these options would work but require extra steps.

I would like to be able to share a link to PDF file on github that would be automatically opened by a browser, instead of being downloaded. It sounds like this is what github.io does but I was really hoping to do this directly from github :)

Thanks,
Rustem

**Reply by Huseyin-Cinar** [reply]
Unfortunately there is no way to do this directly from github because if you can download files in github repos over the internet as a pure file, that is, inside a page.

If we could access it as a pure file, like github.com/example/pdf.pdf, our github repos would be in danger, our secret repos would be stolen, etc. situations could occur.

**Reply by MasterInQuestion** [reply]
E.g. https://github.com/xamidi/pmGenerator/raw/master/pdf/rwth1392_extension_2024.pdf
    Note also: https://github.com/orgs/community/discussions/142728#discussioncomment-11099827

**Comment by rustemos** [comment]
OK, thanks a lot for the explanation. Have a nice weekend!

**Comment by thomassiard** [comment]
Benefits of Displaying PDFs in the Browser:
Enhanced Viewing Options: Users can utilize their browser's built-in PDF viewer, which often includes features like zoom, search, and printing.
Better Navigation: Users can easily switch between tabs and manage multiple documents without losing their place on GitHub.
Accessibility: This option could make it easier for users to access PDF files on different devices and platforms.
Proposal:
Implementing a setting or option that allows users to choose whether to open PDF files in the GitHub window or directly in the browser could be a straightforward solution. This way, users can select their preferred method based on their needs.
